clang/lib/Sema/SemaDecl.cpp

```diff
@@ -1537,6 +1537,10 @@ void Sema::PushOnScopeChains(NamedDecl *D, Scope *S, bool AddToContext) {
       cast<FunctionDecl>(D)->isFunctionTemplateSpecialization())
     return;
 
+  if (isa<UsingEnumDecl>(D) && D->getDeclName().isEmpty()) {
```

Comment:
Why do we push a `UsingEnumDecl` into the scope at all? Would it make sense for the caller to just add the declaration to the `DeclContext` instead of calling `PushOnScopeChains`?

Reply:
`Sema::ActOnUsingEnumDeclaration` invoke `PushOnScopeChains` when `D` is a `UsingEnumDecl` (anonymous or not) and do some extra works.  Just add the declaration would skip checking even it's not anonymous.

Reply:
Are those useful checks? It seems surprising to me that we'd push a `using enum` declaration into scope given that it never introduces any name of its own (the enumerators are handled separately). Are we really performing checks here that we need for a named enum but that are wrong pr unnecessary for an anonymous enum?

Reply:
These checks include name conflict and reserved identifier or not. From this perspective I think it's required here. It will take some time to attempt to skip calling `PushOnScopeChains` and run unittest to find failed cases.

Reply:
For what name? The `using enum` declaration doesn't introduce the name of the enum.

Reply:
Ah, I see. I skip those checking when `D` is `using enum`(`UsingEnumDecl`) by adding
```cpp
if (isa<UsingEnumDecl>(D))
    return;
```
at https://github.com/llvm/llvm-project/blob/cebf77fb936a7270c7e3fa5c4a7e76216321d385/clang/lib/Sema/SemaDecl.cpp#L1529 and it makes sense after runing unit test on `check-clang` locally. Then I think you are right.